src/Compilers/CSharp/Test/Semantic/Semantics/GenericConstraintsTests.cs

```diff
@@ -3270,5 +3268,366 @@ void Test()
                 //         UnmanagedWithInterface(&a);         // fail (does not match interface)
                 Diagnostic(ErrorCode.ERR_GenericConstraintNotSatisfiedValType, "UnmanagedWithInterface").WithArguments("C.UnmanagedWithInterface<T>(T*)", "System.IDisposable", "T", "int").WithLocation(20, 9));
         }
+
+        [Fact]
+        public void UnmanagedGenericStructPointer()
```

Comment:
>UnmanagedGenericStructPointer [](start = 20, length = 29)

Consider testing `IsManagedType` in various symbols would be useful, even though it was not yet made a public API.

Reply:
Will handle this in a future PR